src/components/main/Main.tsx

```diff
@@ -31,16 +38,17 @@ const Main = (): ReactElement => {
     }))
 
     setGenres(updateGenres)
-    setActiveGenresId(updateGenres.find((genre) => genre.active)?.id)
+    filterMovie(updateGenres.find((genre) => genre.active)?.value)
   }
 
   const onSortChange = (id: number, sortingType: SortingType) => {
-    setMovies((movies) => getSortedMovies(movies, id, sortingType))
+    sortMovies(id, sortingType)
   }
 
-  const onOpenMovieDetail = (movie: IMovie) => {
-    dispatch(setActiveMovie(movie))
-  }
+  const moviesCount = useMemo(
+    () => (filteredMovie ? filteredMovie.length : movies.length),
+    [movies.length, filteredMovie],
```

Comment:
why in one case you check length in another all array [movies.length, filteredMovie]?